api/v1alpha1/baseboardmanagement_types.go

```diff
@@ -92,11 +100,51 @@ type BaseboardManagementCondition struct {
 	// Type of the BaseboardManagement condition.
 	Type BaseboardManagementConditionType `json:"type"`
 
+	// Status is the status of the BaseboardManagement condition.
+	// Can be True or False.
+	Status BaseboardManagementConditionStatus `json:"status"`
+
 	// Message represents human readable message indicating details about last transition.
 	// +optional
 	Message string `json:"message,omitempty"`
 }
 
+// +kubebuilder:object:generate=false
+type BaseboardManagementSetConditionOption func(*BaseboardManagementCondition)
+
+// SetCondition updates the Condition if the condition type is present.
+// Appends if new condition is found.
+func (bmj *BaseboardManagement) SetCondition(cType BaseboardManagementConditionType, status BaseboardManagementConditionStatus, opts ...BaseboardManagementSetConditionOption) {
+	currentConditions := bmj.Status.Conditions
+	for i := range currentConditions {
+		// If condition exists, update
+		if currentConditions[i].Type == cType {
+			bmj.Status.Conditions[i].Status = status
+			for _, opt := range opts {
+				opt(&currentConditions[i])
+			}
+			return
+		}
+	}
+
+	// Append new condition to Conditions
+	condition := BaseboardManagementCondition{
+		Type:   cType,
+		Status: status,
+	}
+	for _, opt := range opts {
+		opt(&condition)
+	}
+
+	bmj.Status.Conditions = append(bmj.Status.Conditions, condition)
```

Comment:
The algorithm can be simplified a little.

```suggestion
	var condition *BaseboardManagementCondition
	
	// Check if there's an existing condition.
	for i, c := range bmj.Status.Conditions {
		if c.Type == cType {
			condition = &bmj.Status.Conditions[i]
			break
		}
	}
	
	// We didn't find an existing condition so create a new one and append it.
	if condition == nil {
		condition = &BaseboardManagementCondition{Type: cType}
		bmj.Status.Condtiions = append(bmj.Status.Conditions, condition)
	}
	
	condition.Status = status
	
	for _, opt := range opts {
		opt(condition)
	}
```

Reply:
This is nice! Thank you 👍